### DISCUSSION TITLE
Hooking up a medusa ui component to hook-form

### DISCUSSION THREAD

**Discussion by genox** [medusajs/medusa#5146] [Q&A]
Hi,

I am trying to hook up a `<Select/>` to a form in a modal. The documentation on https://docs.medusajs.com/ui/components/select does not really cover this case (only linking up to local state) and using the same approach as you used in the medusa admin ui codebase doesn't work. I saw that you are not using the same components and have custom implementations for selects, modals, etc.

Are you planning to achieve some kind of feature parity between internal and medusa ui components? Or is there some arcane way to import medusa components (molecules) in a project to have parity?

I noticed that it is not easily possible to achieve the same look and feel for a modal even with the `<Modal>` component, compared to the internal modal component. E.g. there is no footer to add save/cancel buttons. This breaks UX.

Or should we just duplicate code from the main codebase (e.g. https://github.com/medusajs/medusa/tree/develop/packages/admin-ui/ui/src/components/molecules/select?

I would like to get the same UX as the default components, including save/cancel button positions and modal autosizing etc.

I am essentially trying to adapt https://github.com/medusajs/medusa/blob/develop/packages/admin-ui/ui/src/components/forms/product/dimensions-form/index.tsx to my needs.

**Comment by kasperkristensen** [comment]
Hi @genox, we are in the process of updating the entire admin to use the new UI package, it's a large project and will take some time, but over the next many releases you will notice that the different domains will be updated one at a time.

For using the new Select component with `react-hook-form` you will need to use the `Controller` component to wrap the select and control its state, below is an example of how to do that:

```tsx
import { Button, Select } from "@medusajs/ui";
import { useForm, Controller } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";
import * as z from "zod";

const schema = z.object({
  fruit: z.string(),
});

type FormValues = z.infer<typeof schema>;

function App() {
  const form = useForm<FormValues>({
    resolver: zodResolver(schema),
    defaultValues: {
      fruit: "apple",
    },
  });

  const onSubmit = form.handleSubmit((data) => {
    alert(JSON.stringify(data, null, 2));
  });

  return (
    <div className="p-16 min-h-screen bg-ui-bg-base">
      <form onSubmit={onSubmit} className="flex flex-col gap-y-2">
        <Controller
          control={form.control}
          name="fruit"
          render={({ field: { onChange, ...rest } }) => {
            return (
              <Select {...rest} onValueChange={onChange}>
                <Select.Trigger>
                  <Select.Value placeholder="Pick a fruit" />
                </Select.Trigger>
                <Select.Content>
                  <Select.Item value="apple">Apple</Select.Item>
                  <Select.Item value="orange">Orange</Select.Item>
                  <Select.Item value="melon">Melon</Select.Item>
                </Select.Content>
              </Select>
            );
          }}
        />
        <div className="flex items-center justify-end w-full">
          <Button type="submit">Submit</Button>
        </div>
      </form>
    </div>
  );
}

export default App;
```
[CodeSandbox](https://codesandbox.io/p/sandbox/red-moon-lmp7hv?file=%2Fsrc%2FApp.tsx%3A1%2C1-54%2C1)

When I get the time I will update the UI docs with examples of how to hook things up with `react-hook-form` for forms, `@tanstack/react-table` for tables, etc. But I hope the above example can help you in the meantime.

In regards to the Modal, then it's no longer part of our design system, and instead we have [Drawer](https://docs.medusajs.com/ui/components/drawer) and [FocusModal](https://docs.medusajs.com/ui/components/focus-modal). The Drawer component will be used everywhere we currently use a Modal going forward, and FocusModal is already an existing component but it has a slightly new look in `@medusajs/ui`.

**Reply by pevey** [reply]
I'm not sure if you linked the right codesandbox example above

**Reply by kasperkristensen** [reply]
Link should be good, what do you see when you open it @pevey? Maybe I just don't know how to share sandboxes correctly, in any case I have added the code example to answer.

**Reply by pevey** [reply]
The link I clicked didn't seem to have that example.  It had just a Table example.  I could have been looking in the wrong files.

**Reply by kasperkristensen** [reply]
Ahh okay, yea it’s in the App.tsx file
